spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/GcpContextAutoConfiguration.java

```diff
@@ -40,8 +41,10 @@
  * @author Mike Eltsufin
  * @author Elena Felder
  * @author Chengyuan Zhao
+ * @author Serhat Soydan
  */
 @Configuration
+@ConditionalOnProperty(name = "spring.cloud.gcp.core.enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
We should also add it to `spring-cloud-gcp/spring-cloud-gcp-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json`.